### ISSUE TITLE
Archeologist quirks, CS gifted units spawn, DP exceeding DOF

### ISSUE THREAD

**Issue by kaspergm** [LoneGazebo/Community-Patch-DLL#2563]
_1. Mod version (i.e Date - 4/23):_
CP 09-18

_2. Mod list (if using Vox Populi only, leave blank):_
Various whoward mods.

_3. Error description:_ 
Two bugs + one possible bug noticed in otherwise succesfull game:

1) There are problems with Archeologists sometimes but not always being allowed to dig in foreign lands. Sometimes it works normally, sometimes there is no actions menu at all for the archaeologist, sometimes the actions menu is there but option to dig greyed out with text that it cannot be used in foreign lands. Related issue: Many AI archaeologists seem to just stand on the dig site doing nothing. One AI player did have artifacts to trade, however, and hence must have been able to dig up artifacts (that was Germany, btw., same AI player whos lands I could dig in - I wonder whether that is significant). I posted some screenshots below.

2) When receiving a unit from a miliatry city state on a different continent, the unit spawns within the CS lands rather than in your own lands. This is the case both for military units and great people. However city states on your own continents do not have this issue.

3) Not sure if this is a bug, but: Two AI players had DOF and DP (defensive pact). They end their DOF after taking different ideologies. Relationship goes very bad between them. However the defensive pact remains. That seems strange. Not sure if it's intentional or a bug.

_4. Screenshots (optional):
http://s105.photobucket.com/user/kaspergm/media/Arch1.jpg.html
http://s105.photobucket.com/user/kaspergm/media/Arch2.jpg.html
http://s105.photobucket.com/user/kaspergm/media/Arch3.jpg.html

**Comment by LoneGazebo** [comment]
Thanks!

**Comment by LoneGazebo** [comment]
@ilteroi mind looking at archaeologists? Might be a unit mission issue.

**Comment by Iamblichos** [comment]
I believe you need open borders in foreign lands for the interface mode to kick in for Archaeologists.

**Comment by LoneGazebo** [comment]
Well beyond that they seem to be acting up for automated archaeologists as well.

**Comment by kaspergm** [comment]
Obviously they had opened their borders to me, otherwise I would not have been able to get the archaeologist in in the first place. I might not have opened my borders to them, so technically that could account for some of it. Still there'a also the odd case with city state.

**Comment by kaspergm** [comment]
No Rival Territory promotion when playing CP only.

**Comment by ilteroi** [comment]
1 - if one archeologist is digging the other one in the same plot is blocked, so that maybe explains your screenshot nr 3.

observations from digging in the code:
- no rival territory promotion for archeologists
- there's a diplo method which don't seem to be used CvDiplomacyAI::IsPlayerAgreeNotToDig(), instead there's an alternative method CvDiplomacyAI::IsPlayerMadeNoDiggingPromise(). which is used. weird.
- the the dig itself is a "build", id 29, with the corresponding improvement 24. the checks for what can built and under which conditions are extensive and hard to follow. but there's one bit:
```	// In general, cannot clear features in the lands of players that aren't on our team or that we're at war with
```
however, there's an exception if the flag GC.getImprovementInfo(eImprovement)->IsIgnoreOwnership() is set.
- there was also an effect where automated archeologists would start and dig and then move on before finishing. is fixed.

2 - reproduced, but is it a bug? not sure.

**Comment by kaspergm** [comment]
Just for clarification, the Egyptian archaeologist was not performing a dig. He was just standing on the tile (for many turns) doing nothing.

With regards to the abilities to dig on improved tile and hence remove improvements: In unmodded civ, you can always dig on improved tiles in foreign lands (although civs will dislike you for it, even more so if you made the promise not to dig). This seems to be changed in CP where for many versions I have not been able to dig if site is on an improved tile (not sure if that was intentional or not). However in this game, I [i]could[/i] excavate the german sites in spite of them sitting on top of farms.

**Comment by Der-Zorn-Gottes** [comment]
I just remebered another issue with Archeologist that I wanted to report but forgot. If you want me to make a separate thread, I will.
Last patch version, I often found sites "pre-digged" by the AI. So my Archeologist would move on the tile, and it only takes him 7 turns instead of 14 to dig it up. It has nothing to do with workers having pre-chopped forest or jungle, it was also on flatland. 
What must have happened is that the AI removed their Archeologist while he was digging.
Confirmation for this suspicion was that I checked spots that were blocked by a AI Archeologist a couple of turns later, and indeed it was free (but still not completely excavated).
